lib/rubocop/cop/style/documentation.rb

```diff
@@ -70,6 +75,26 @@ def associated_comment?(node, ast_with_comments)
           # annotation, it's OK.
           ast_with_comments[node].any? { |comment| !annotation?(comment) }
         end
+
+        # The :nodoc: comment is not actually associated with the class/module
+        # ifself but its first commentable child node. Unless the element is
+        # tagged with :nodoc:, the search proceeds to check its ancestors for
+        # :nodoc: all.
+        def nodoc?(node, ast_with_comments, require_all = false)
+          return false unless node
+          nodoc_node = node.children.last
+          return false unless nodoc_node
+
+          nodoc_node = nodoc_node.children.first while nodoc_node.type == :begin
+          comment = ast_with_comments[nodoc_node].first
+
+          if comment && comment.loc.line == node.loc.line
+            regex = /^#\s*:nodoc:#{"\s+all\s*$" if require_all}/
```

Comment:
Shouldn’t the `\s*$` part can be moved out of the string interpolation, so the acceptable regexes are `/^#\s*:nodoc:\s*$/` and `/^#\s*:nodoc:\s+all\s*$/`, and *not* `/^#\s*:nodoc:/`?